### PR TITLE
ACSP-222- Improve ingestion webhook

### PR DESCRIPTION
**Pull request by rojoangel**
## Description

Improve ingestion webhook by
* removing some unneeded code
* checking the response when sending events and respond HTTP/400 when the event type is unknown (instead of HTTP/200)

When the event passed was unknown to the system, the ingestion webhook responded with an HTTP/200 status code before this change

```
{
	"response": {
		"message": "Event published successfully",
		"success": true
	},
	"type": "external_provider.customer_created.v1.0"
}
```
After this change it'll respond with HTTP/400 status code
```
{
	"error": "Unable to publish event external_provider.customer_created.v1.0: Unknown event type"
}
```

## Motivation and Context

With this change, the ingestion webhook code 
* is cleaner
* surfaces the fact that there's been error when processing a request caused by an unknown event type

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: actions/ingestion/webhook/index.js [hunk 1]
```diff
@@ -76,22 +68,27 @@ async function main (params) {
 
     logger.info('Process event data')
     logger.debug(
-          `Process event ${event.type} for entity ${event.entity}`)
+          `Process event ${params.data.event}`)
```
Comment (pablomoreno61) [inline actions/ingestion/webhook/index.js:71]: is it in a different line on purpose?

File: actions/ingestion/webhook/index.js [hunk 2]
```diff
@@ -76,22 +68,27 @@ async function main (params) {
...
+    logger.debug(`Publish event result: ${publishEventResult}`)
+    if (publishEventResult !== PUBLISH_EVENT_SUCCESS) {
+      logger.error(`Unable to publish event ${params.data.event}: Unknown event type`)
+      return errorResponse(HTTP_BAD_REQUEST, `Unable to publish event ${params.data.event}: Unknown event type`)
+    }
 
     logger.info(`Successful request: ${HTTP_OK}`)
 
-    return successResponse(event.type, {
+    return successResponse(params.data.event, {
```
Comment (pablomoreno61) [inline actions/ingestion/webhook/index.js:91]: params.data.event is accessed a lot, maybe makes sense to assign it to a variable to make it more readable

**Review by pablomoreno61 — APPROVED** [review]

File: actions/ingestion/webhook/index.js (hunk 1 above)
Comment (rojoangel) [reply actions/ingestion/webhook/index.js:71]: Good point! I will remove the duplication

File: actions/ingestion/webhook/index.js (hunk 2 above)
Comment (rojoangel) [reply actions/ingestion/webhook/index.js:91]: Sure!

**Pull request merged**